routes/dashboard/portals/project-allotment-student/index.js

```diff
@@ -13,13 +13,32 @@ var adminsModel = fq("schemas/admins");
 var projectsModel = fq("schemas/projects");
 var applicationsModel = fq("schemas/project-applications");
 
+var forbidden_batches = [];
+settingsModel.find(
+  {
+    name: "proj-allotment-forbidden-batches"
+  },
+  function(err, docs) {
+    if (docs[0]) {
+      forbidden_batches = docs[0].value;
+    }
+  }
+);
+function forbidOrNot(email) {
+  let allow = false; // don't forbid
+  forbidden_batches.forEach(batch => {
+    if (email.includes(batch)) allow = true;
```

Comment:
Use `email.indexOf(batch) == 1` as the current implementation may cause issues in future if we have ID numbers greater than 2000. Sample email which may cause issues when batch 2022 is forbidden but 2020 is allowed `f20202022@hyderabad.bits-pilani.ac.in`. 

An alternative (maybe better) implementation is `req.user.idNumber.indexOf(batch) == 0`